### ISSUE TITLE
Back button image not showing in Stack navigator (although button is there and pressable) - iOS and Android

### ISSUE THREAD

**Issue by sergioisidoro** [react-navigation/react-navigation#11792]
### Current behavior

I'm having an issue where the arrow image of the back button does not show on Android (Expo Go - simulator and Device), despite the button being there, and being pressable. 

The image shows on iOS, and it's only an issue on a nested navigator, not the Root stack navigator
```
AppNav (Stack) -- Back button visible
   Drawer Nav (Drawer)
       User Settings Screen
       Project Navigator (Stack)   
         Tab navigator
             Screen 1
             Screen 2
         Stack Screen 1 -- Back button there, but not visible
```

* Because the button is visible on the Root stack I don't think this is an issue with the Assets imports / loading
*  It also doesn't' seem to be an issue with the header styles, since if I move `Stack screen 1` to the root stack, the back button starts showing.
* Inspecting the elements shows correctly the touchables, and the correct elements. The element is clickable, and does go back if pressed. It's just the arrow that is not visible. 
* I have removed all header styles, and all other headers are `headerShown: false`.  Nothing helped
* Because the button is there and is showing, it is not a problem of the stack being empty (with nothing to go back to)
* Adding a `headerBackImage` bypasses the issue

<img width="692" alt="Screenshot 2024-01-18 at 17 30 04" src="https://github.com/react-navigation/react-navigation/assets/5014629/764aa4a7-2d63-4358-ae0a-e7264bacebf2">

iOs (left), Android (right)

NOTE: Cannot reproduce in Expo snack because of Reanimated issues

### Expected behavior

Back button showing.

### Reproduction

https://snack.expo.dev/@smaisidoro/talking-violet-coffee

### Platform

- [X] Android
- [X] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [ ] @react-navigation/material-top-tabs
- [ ] @react-navigation/stack
- [X] @react-navigation/native-stack
- [ ] react-native-tab-view

### Environment

Expo Go

```
    "@react-navigation/bottom-tabs": "^6.5.11",
    "@react-navigation/drawer": "^6.6.6",
    "@react-navigation/native": "~6.1.9",
    "@react-navigation/native-stack": "^6.9.17",
    "@react-navigation/stack": "~6.3.20",
```

Similar issues found:
https://github.com/react-navigation/react-navigation/issues/8609
https://github.com/react-navigation/react-navigation/issues/11657

Worked around the issue by providing a custom back button image as a `headerBackImage` component

**Comment by oliverdolgener** [comment]
I face the same issue but on iOS 😕
Looks as if it would only affect my production builds. On the simulator it seems to be there

**Comment by bnussman** [comment]
I'm having a similar issue to that @oliverdolgener. Everything is fine on Expo Go and in an Expo Dev Client, but my drawer icon is missing on production builds for iPhone only. It shows up on iPad somehow

**Comment by RyanTG** [comment]
I'm experiencing the same as @oliverdolgener - back button is missing on iOS production-only, though not Android.

This occurred after upgrading Expo to SDK 50. 

I experienced it with this set of react navigation packages: 

```
    "@react-navigation/bottom-tabs": "^6.3.3",
    "@react-navigation/drawer": "^6.4.4",
    "@react-navigation/elements": "^1.3.21",
    "@react-navigation/native": "^6.0.12",
    "@react-navigation/stack": "^6.2.3",
```

as well as with 

```
    "@react-navigation/bottom-tabs": "^6.5.11",
    "@react-navigation/drawer": "^6.6.6",
    "@react-navigation/elements": "^1.3.21",
    "@react-navigation/native": "^6.1.9",
    "@react-navigation/stack": "^6.3.20",
```

I am not using OTA updates, as another noted above. 

Not sure how to proceed here.

**Comment by oliverdolgener** [comment]
My current workaround is to use a custom back icon. But would be nice to have the default ones working again

**Comment by TMcLoone** [comment]
Also having this issue after Expo 50 updgrade

**Comment by fgauna12** [comment]
Same issue for me on iOS and with stack navigator. Only happens in production builds. 

```
    "@react-navigation/bottom-tabs": "^6.5.11",
    "@react-navigation/native": "^6.1.7",
    "@react-navigation/stack": "^6.1.7",
```

**Comment by carbonatedcoder** [comment]
Has there been any progress on this issue or a known cause? After upgrading to Expo SDK 50, all our iOS production builds are missing the back button. It seems to still be pressable, but almost like the icon asset is just missing from the build. The only workaround we've found is that publishing an OTA subsequent to the build makes the back button appear again.

It does not happen within Expo Go.

**Packages being used:**
```
    "@react-navigation/bottom-tabs": "^6.5.7",
    "@react-navigation/elements": "^1.3.17",
    "@react-navigation/native": "^6.1.6",
    "@react-navigation/stack": "^6.3.16",
```

**Comment by Eonfuzz** [comment]
> Here i my temporary fix. Expo will not be in a hurry to fix the issue, they have a new favourite :-) : `<Stack.Navigator screenOptions={{ headerShown: true, headerBackImage: () => <BackIcon /> }} initialRouteName="Welcome" >` Then the back icon: `const BackIcon = props => { return <Icon name="chevron-left" size={40} {...props} /> }`

Thanks mate, had the same issue over here and it blindsided us. Probably worth having the icons be different per platform but otherwise worked

**Comment by dylanrandle** [comment]
> Has there been any progress on this issue or a known cause?
> 
> After upgrading to Expo SDK 50, all our iOS production builds are not showing the back button after a fresh build. It is still pressable, but it's like the icon asset is missing or something. The only workaround we've found is that publishing an OTA subsequent to the build makes the back button appear again. It does not happen within Expo Go.
> 
> When building a simulator preview build and inspecting the application bundle using "Show Package Contents", I see the back icon bundled into the build at `assets/node_modules/@react-navigation/elements/src/assets/back-icon.png`, yet it does not show in app. Only after the first OTA does it start to appear.
> 
> **Packages being used:**
> 
> ```
>     "@react-navigation/bottom-tabs": "^6.5.7",
>     "@react-navigation/elements": "^1.3.17",
>     "@react-navigation/native": "^6.1.6",
>     "@react-navigation/stack": "^6.3.16",
> ```

I am seeing the exact same issue as @carbonatedcoder 
- iOS builds only
- Confirmed that the `back-icon@1x.ios.png`, `back-icon@2x.ios.png`, and `back-icon@3x.ios.png` are included in the build (although, it's missing the `back-icon@1.5x.ios.png` and `back-icon@4x.ios.png` which are included in the original code). 
- "Fixed" by pushing OTA update

**Comment by dylanrandle** [comment]
@EvanBacon can you or someone at Expo to take a look at this?

**Comment by outaTiME** [comment]
Exactly, I am experiencing the same problem after migrating to Expo 50, the back icon files of react navigation are not resolved in the iOS production builds.

For example, the following code:

```js
import { useAssets } from 'expo-asset';
const [assetsLoaded, assetsError] = useAssets([
  require('@react-navigation/elements/src/assets/back-icon.png'),
  require('@react-navigation/elements/src/src/assets/back-icon-mask.png'),
]);
if (assetsError) {
  console.warn(assetsError);
}
```

throws the following error:

```
{ [Error: File './ -- file:///' could not be copied to 'file:///var/mobile/Containers/Data/Application/9B4B3B09-F681-4578-B13F-7EBA52903F5B/Library/Caches/ExponentAsset-0ea69b5077e7c4696db85dbcba75b0e1.png'] code: 'E_FILE_NOT_COPIED' }
```

**Comment by jzhangdev** [comment]
Has anyone solved this problem? Expo 51  still have the same problem on ios production build.

**Comment by WhidRubeld** [comment]
I confirm. This problem occurs in almost every build for iOS when using Expo 50

**Comment by dylanrandle** [comment]
@sergioisidoro Would you be willing to update the description to include iOS? If not, we can create a new issue since many of us are only seeing this on iOS.

**Comment by sergioisidoro** [comment]
@dylanrandle Done! Thanks for bringing attention to this

I also removed the word nested because it seems people are getting this issue on non nested navigators (?) **can please someone confirm that it is indeed the case?** or does this only happen on nested navigators?

Also for people finding this issue for the first time, note that it can be worked around by providing a custom image through the prop.

**Comment by WhidRubeld** [comment]
I confirm that on any Stack navigation screen. It doesn't matter if it's nested or in Root.

```
    "@react-navigation/elements": "^1.3.19",
    "@react-navigation/native": "^6.1.8",
    "@react-navigation/native-stack": "^6.9.14", // not used, used only JS stack navigator
    "@react-navigation/stack": "^6.3.18",
```

**Comment by thelamedia** [comment]
I am also still having this issue on iOS production builds in SDK 51.0.11.  RN elements, native, stack and bottom tabs are all the latest version, updated using expo install. 
Dev (simulator and client) and android all work properly.

**Comment by tomerittah** [comment]
I've used the following workaround which is I basically took from `HeaderBackButton` of `react-navigation/elements` for `IOS` and changed the `Image` to `Icon` and it worked for production `EAS` build:
```
import * as React from 'react'
import { StyleSheet, I18nManager } from 'react-native'
import { Icon } from 'react-native-elements'

const styles = StyleSheet.create({
  icon: {
    height: 21,
    width: 13,
    marginLeft: 8,
    marginRight: 22,
    marginVertical: 12,
    transform: [{ scaleX: I18nManager.isRTL ? -1 : 1 }],
  },
})

// TODO: Delete once the missing header back button is solved
const HeaderBackImage = ({ tintColor }): React.ReactNode => {
  return (
    <Icon name="chevron-left" color={tintColor} type="font-awesome-5" iconStyle={styles.icon} />
  )
}

export default HeaderBackImage
```

Then:
```
<Stack.Navigator
      screenOptions={{
        ...
        headerBackImage: HeaderBackImage,
      }}
    >
    ...
</Stack.Navigator>
```

**Comment by rikur** [comment]
Wow, #12019 really fixes the issue. I'm currently patching it with patch-package:

```
diff --git a/node_modules/@react-navigation/elements/src/assets/back-icon@1.5x.android.png b/node_modules/@react-navigation/elements/src/assets/back-icon@1.5x.android.png
deleted file mode 100644
index ad03a63..0000000
Binary files a/node_modules/@react-navigation/elements/src/assets/back-icon@1.5x.android.png and /dev/null differ
diff --git a/node_modules/@react-navigation/elements/src/assets/back-icon@1.5x.ios.png b/node_modules/@react-navigation/elements/src/assets/back-icon@1.5x.ios.png
deleted file mode 100644
index 1fa30ec..0000000
Binary files a/node_modules/@react-navigation/elements/src/assets/back-icon@1.5x.ios.png and /dev/null differ
```

i.e. just delete the node_modules/@react-navigation/elements/src/assets/back-icon@1.5*.png files.

**Comment by summerkiflain** [comment]
> Wow, #12019 really fixes the issue. I'm currently patching it with patch-package:
> 
> ```
> diff --git a/node_modules/@react-navigation/elements/src/assets/back-icon@1.5x.android.png b/node_modules/@react-navigation/elements/src/assets/back-icon@1.5x.android.png
> deleted file mode 100644
> index ad03a63..0000000
> Binary files a/node_modules/@react-navigation/elements/src/assets/back-icon@1.5x.android.png and /dev/null differ
> diff --git a/node_modules/@react-navigation/elements/src/assets/back-icon@1.5x.ios.png b/node_modules/@react-navigation/elements/src/assets/back-icon@1.5x.ios.png
> deleted file mode 100644
> index 1fa30ec..0000000
> Binary files a/node_modules/@react-navigation/elements/src/assets/back-icon@1.5x.ios.png and /dev/null differ
> ```
> 
> i.e. just delete the node_modules/@react-navigation/elements/src/assets/back-icon@1.5*.png files.

I had to remove these files from other `lib/module` & `lib/commonjs` folders also before it started working for me in expo development build.

**Comment by pfcodes** [comment]
Confirmed happening in Expo 51. Development build works fine, but as soon as a TestFlight/Production build is built it's invisible.

**Comment by Darshan562** [comment]
> I'm having a similar issue to that [@oliverdolgener](https://github.com/oliverdolgener). Everything is fine on Expo Go and in an Expo Dev Client, but my drawer icon is missing on production builds for iPhone only. It shows up on iPad somehow
> 
> Edit: Turns out I was at my expo OTA update limit, which cased my app to not push new icons to the client. Had to pay for an upgraded Expo Updates limit.

I face the exact same issue for my android i see the back arrow button but for the iOS app i only see the Back text nt the arrow so how you fix this issue?
For better understanding i attach the images

![Image](https://github.com/user-attachments/assets/f18a19b0-851d-4d7f-a6a6-058264ec3861)

**Comment by JonasHiltl** [comment]
I'm still having this issue on `expo-router@~6.0.15` and also on `web`. I fixed this by copying the `back-icon....png` to my assets folder manually and specifying 
```
headerBackImageSource: require('../../../assets/images/back-icon.png'),
```

**Comment by harterc1** [comment]
I'm seeing this currently on expo-router 6.0.21 on expo 54. so, nearly the latest version. I don't see any indication it'll be fixed in expo 55

**Comment by sergioisidoro** [comment]
I think yeah, it's safe to assume this may not be fixed at all. Could it be that 1.5x device pixel density are not common enough that this deserves the propper attention?

But I think particular issue has been fixed if you have the latest `@react-navigation/elements` installed, which completely removes the 1.5x versions of the back icons. 

At least I don't seen anymore any 1.5x assets in main: https://github.com/react-navigation/react-navigation/tree/e14137884d6b011d1bf79b3c2fde076738ac4406/packages/elements/src/assets

**Comment by harterc1** [comment]
I've been testing all day with the latest expo-doctor results and have not seen the issue.

```   
"expo": "~54.0.33",
"expo-router": "~6.0.23",
```
